lib/src/json_schema/validator.dart

```diff
@@ -795,6 +846,30 @@ class Validator {
 
   int get _evaluatedItemCount => _evaluatedItemsContext.lastOrNull;
 
+  //////
+  // Helper functions to deal with unevaluatedProperties.
+  //////
+
+  _pushEvaluatedPropertiesContext() {
+    _evaluatedPropertiesContext.add(Set<Instance>());
+  }
+
+  _popEvaluatedPropertiesContext() {
+    var last = _evaluatedPropertiesContext.removeLast();
+    if (_evaluatedPropertiesContext.isNotEmpty) {
+      _evaluatedPropertiesContext.last.addAll(last);
+    }
+  }
+
+  bool get _isInEvaluatedPropertiesContext => _evaluatedPropertiesContext.isNotEmpty;
+
+  _addEvaluatedProp(Instance i) {
+    if (_evaluatedPropertiesContext.isNotEmpty) {
```

Comment:
Is doing nothing the right thing to do otherwise? Discount double-check

Reply:
Yes, doing nothing is the correct thing when the list is empty. This allows me to sprinkle this helper method in areas the property is validated and not care of we're in a so called `evaluatedPropertiesContext` or not.